Guide.docc/FeatureMigration.md

```diff
@@ -0,0 +1,97 @@
+# Migrating to upcoming language features
+
+Migrate your project to upcoming language features.
+
+Upcoming language features can be enabled in the Swift compiler via a `-enable-upcoming-feature
+<FeatureName>` flag. Some of these features also support a migration mode, which will output
+diagnostics with fix-its to allow migrating to that mode. The exact semantics of such a migration is
+dependent on the feature, see their [corresponding
+documentation](https://docs.swift.org/compiler/documentation/diagnostics#Upcoming-language-features-list)
+for more details.
+
+## SwiftPM
+
+> Note: This feature is in active development. Test with a [nightly
+> snapshot](https://www.swift.org/install) for best results.
+
+`swift package migrate` builds and applies migration fix-its to allow for semi-automated migration.
+Make sure to be in a clean state, since applying the fix-its will modify files in the package *in
```

Comment:
```suggestion
Make sure to start in a clean state, since applying the fix-its will modify files in the package *in
```

Perhaps? We might also want to explicitly say what we mean by a "clean state", and that your project builds successfully before you migrate.

Reply:
Good point, thank you 🙇‍♂️ Update this to:
> Make sure to start with a clean working tree (no current changes staged or otherwise) and a working
> build - applying the fix-its requires there to be no build errors and will modify files in the
> package *in place*.